File: sonarqube-companion-rest/src/main/java/pl/consdata/ico/sqcompanion/history/ProjectViolationsHistoryDiff.java
package pl.consdata.ico.sqcompanion.history;

import lombok.Builder;
import lombok.Data;

import java.time.LocalDate;

@Data
@Builder
public class ProjectViolationsHistoryDiff {

    private final String projectKey;
    private final String projectId;
    private final Violations violationsDiff;
    private final LocalDate fromDate;
    private final LocalDate toDate;

    public String getFromDateString() {
        return fromDate.toString();
    }

    public String getToDateString() {
        return toDate.toString();
    }

}

File: sonarqube-companion-frontend/src/app/violations/project-violations-history-diff.ts
import {Violations} from './violations';

export class ProjectViolationsHistoryDiff {

  violations: Violations;
  addedViolations: Violations;
  removedViolations: Violations;
  fromDate;
  toDate;

  constructor(data) {
    this.violations = new Violations(data.violationsDiff);
    this.addedViolations = new Violations(data.addedViolations);
    this.removedViolations = new Violations(data.removedViolations);
    this.fromDate = data.fromDateString;
    this.toDate = data.toDateString;
  }

}
